packages/wallets/ledgerconnect/src/adapter.ts

```diff
@@ -0,0 +1,236 @@
+
+    private _connecting: boolean;
+    private _wallet: LedgerConnectWallet | null;
+    private _publicKey: PublicKey | null;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' || typeof document === 'undefined'
+            ? WalletReadyState.Unsupported
+            : WalletReadyState.NotDetected;
+
+    constructor(_config: LedgerConnectWalletAdapterConfig = {}) {
+        super();
+        this._connecting = false;
+        this._wallet = null;
+        this._publicKey = null;
+        if (this._readyState !== WalletReadyState.Unsupported) {
+            const handler = (event: MessageEvent<any>) => {
+                if (typeof event.data === 'object' && event.data.__ledgerConnect_loaded) {
+                    if (this._readyState !== WalletReadyState.Installed) {
+                        this._readyState = WalletReadyState.Installed;
+                        this.emit('readyStateChange', this._readyState);
+                    }
+                    window.removeEventListener('message', handler);
+                }
+            };
+
+            window.addEventListener('message', handler);
+
+            scopePollingDetectionStrategy(() => {
+                if (window.ledgerConnectSolana?.isLedgerConnect) {
+                    window.removeEventListener('message', handler);
+                    if (this._readyState !== WalletReadyState.Installed) {
+                        this._readyState = WalletReadyState.Installed;
+                        this.emit('readyStateChange', this._readyState);
+                    }
+                    return true;
+                }
+                return false;
+            });
+        }
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._wallet?.isConnected;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Installed) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.ledgerConnectSolana!;
```

Comment:
Will we be defining this on LLM side?
Who declares and initializes this in the first place?

Couldn't we have something that's not completely tied to Solana in the global scope, something like `ledgerConnect.Solana` instead?

Reply:
The interface is defined [here](https://github.com/harrisonf-ledger/wallet-adapter/blob/f3fb6c94d6d6a63a4f2a8fb12f63b1dc4bd9e193/packages/wallets/ledgerconnect/src/adapter.ts#L24), and our extension will initialize and set this value.

We're using `window.{product_name}Solana` because that's the convention I've seen the repo owners suggest everyone start using going forward. It used to be `window.solana`.

Reply:
Thanks ! But is it a good idea to have the definition of `LedgerConnectWallet` on Solana's wallet-adapter side and have our extension instantiate it using the definition from Solana's wallet-adapter ?